src/components/SortDropdown.js

```diff
@@ -0,0 +1,129 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { makeStyles, withStyles } from '@material-ui/core/styles';
+import ExpandLessIcon from '@material-ui/icons/ExpandLess';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import FormControl from '@material-ui/core/FormControl';
+import InputBase from '@material-ui/core/InputBase';
+import NativeSelect from '@material-ui/core/NativeSelect';
+
+const BootstrapInput = withStyles((theme) => ({
+  input: {
+    borderRadius: '4px 4px 0px 0px',
+    position: 'relative',
+    backgroundColor: theme.palette.background.paper,
+    minWidth: '177px',
+    minHeight: '43px',
+    lineHeight: '18px',
+    padding: '7px 10px 7px 10px',
+    fontWeight: 'bold',
+    fontFamily: 'Work Sans',
+  },
+}))(InputBase);
+
+const useStyles = makeStyles((theme) => ({
+  formStyle: {
+    borderRadius: '4px 4px 0px 0px',
+  },
+  hiddenStyle: {
+    display: 'none',
+  },
+  dropdownOptionStyle: {
+    fontWeight: 'bold',
+  },
+}));
+
+export default function SortDropdown({
+  inputSortMethodList,
+  defaultSortMethod,
+  setSortMethod,
+}) {
+  const classes = useStyles();
+  const [isDropdownExpand, setIsDropdownExpand] = useState(false);
+  const sortMethodList = inputSortMethodList;
+  const [selectedSortMethod, setSelectedSortMethod] =
+    useState(defaultSortMethod);
+  const [sortMethodMap] = useState({
+    'best match': 'Best Match',
+    'updated': 'Last Updated',
+    'stars': 'Stargazer Count',
+  });
+  const [clickedOutside, setClickedOutside] = useState(false);
+  const myRef = useRef();
+
+  useEffect(() => {
```

Comment:
Is the NativeSelect unable to respond in the way we want it to when focus is lost (i.e. click anywhere but the select dropdown).

Reply:
Since the NativeSelect component is not able to reset the dropdown icon status when focus is lost, we need to add the useRef logic to handle that.

NativeSelect component example sandbox: https://codesandbox.io/s/g4fy3?file=/demo.js